### PR TITLE
MediumStrenght generator rely on deprecated mcrypt extension

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes (maybe -- partial 9.2 only)
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2393

### CODE REVIEW

File: inc/toolbox.class.php [hunk 1]
```diff
@@ -1477,20 +1477,13 @@ static function getItemTypeTabsURL($itemtype, $full = true) {
...
    **/
-   static function getRandomString($length, $high = false) {
+   static function getRandomString($length) {
 
       $factory = new RandomLib\Factory();
-      if ($high) {
-         /* Notice "High" imply mcrypt extension, unwanted for now
-            See https://github.com/ircmaxell/RandomLib/issues/57 */
-         $generator = $factory->getMediumStrengthGenerator();
-      } else {
```
Comment (remicollet) [inline inc/toolbox.class.php]: please keep the link to upstream issue

**Comment by remicollet** [comment]
Really needed ?

**Comment by trasher** [comment]
On the core side, this was unit tested without any issues; but one of the plugins get a deprecated issue using the `User::getToken()` method from the master branch :/
So, I guess there really is an issue somewhere

File: inc/toolbox.class.php (hunk 1 above)
Comment (trasher) [inline inc/toolbox.class.php]: Re-added

**Comment by remicollet** [comment]
Missing openssl extension ?

**Comment by trasher** [comment]
I don't know, I've admitted travis environment were the same :)

**Comment by btry** [comment]
Hi

My plugin requires openssl extension and checks it in requirements. The installation of the plugin is unit tested and it passes.

**Comment by trasher** [comment]
The issue is reproductible on GLPI side with debug mode activated (error_reporting(E_ALL | E_STRICT) and with mcrypt module installed.

Issue is from the RandomLib library that relies on mcrypt for "medium" and "high"...

Maybe https://stackoverflow.com/questions/4356289/php-random-string-generator/31107425#31107425 would be an alternative to the lib?

**Comment by trasher** [comment]
See #2423

**Pull request closed without merging**